docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -122,13 +127,18 @@ public String toString() {
               .add("commands", commands)
               .add("volumes", volumes)
               .add("env", env)
+              .add("portBindings", portBindings)
               .toString();
    }
 
    public static final DockerTemplateOptions NONE = new DockerTemplateOptions();
 
    public DockerTemplateOptions volumes(Map<String, String> volumes) {
-      this.volumes = Optional.<Map<String, String>> of(ImmutableMap.copyOf(volumes));
+      for (Map.Entry<String, String> entry : checkNotNull(volumes, "volumes").entrySet()) {
+         checkNotNull(entry.getKey(), "host paths must not be null");
+         checkNotNull(entry.getValue(), "container paths must not be null");
+      }
```

Comment:
I couldn't remember. So, do you think we get a useful enogh error when we create an `ImmutableMap` from a `HashMap` with nulls in it? If so, we just need the `checkNotNull(input)` type validation, and I'll update the PR appropriately. _**Checked** and it does a `checkEntryNotNull` so we're all good_